src/util/symbol.js

```diff
@@ -129,6 +129,79 @@ var Pin = graphic.extendShape({
     }
 });
 
+/**
+ * Sausage: similar to sector, but have half circle on both sides
+ * @inner
+ */
+export var Sausage = graphic.extendShape({
+
+    type: 'sausage',
+
+    shape: {
+
+        cx: 0,
+
+        cy: 0,
+
+        r0: 0,
+
+        r: 0,
+
+        startAngle: 0,
+
+        endAngle: Math.PI * 2,
+
+        clockwise: true
+    },
+
+    brush: graphic.fixClipWithShadow(function () {
+        return this.startAngle === this.endAngle;
+    }),
+
+    buildPath: function (ctx, shape) {
+        var x = shape.cx;
+        var y = shape.cy;
+        var r0 = Math.max(shape.r0 || 0, 0);
+        var r = Math.max(shape.r, 0);
+        var dr = (r - r0) * 0.5;
+        var rCenter = r0 + dr;
+        var startAngle = shape.startAngle;
+        var endAngle = shape.endAngle;
+        var clockwise = shape.clockwise;
+
+        var unitStartX = Math.cos(startAngle);
+        var unitStartY = Math.sin(startAngle);
+        var unitEndX = Math.cos(endAngle);
+        var unitEndY = Math.sin(endAngle);
+
+        var lessThanCircle = clockwise
+            ? endAngle - startAngle < Math.PI * 2
+            : startAngle - endAngle < Math.PI * 2;
+
+        if (lessThanCircle) {
+            ctx.moveTo(unitStartX * r0 + x, unitStartY * r0 + y);
+
+            ctx.arc(unitStartX * rCenter + x, unitStartY * rCenter + y, dr,
+                -Math.PI + startAngle, startAngle, !clockwise);
```

Comment:
Maybe it is not important but I still prefer to point out the code style issue about the line wrapping:

By google code style the it recommends "at least double normal indent" [1][2]. That is, if the normal is 2 space, the double should be 4 space. If the normal is 4 space, the double should be 8 space. So here should be:
```js
ctx.arc(unitStartX * rCenter + x, unitStartY * rCenter + y, dr,
        -Math.PI + startAngle, startAngle, !clockwise);
```

By ecomfe code style (where most of the echarts code style come from), the back parenthesis is recommended to placed in a separate line (align with the indent of the line where the start parenthesis is) and no need to double the indent. So the line wrap should be:
```js
ctx.arc(unitStartX * rCenter + x, unitStartY * rCenter + y, dr,
    -Math.PI + startAngle, startAngle, !clockwise
);
```

Both of the two styles are reasonable (to distinguish between the line break and deeper block of code). But the current way is not reasonable enough I think.

[1] https://google.github.io/styleguide/jsguide.html#formatting-indent
[2] https://google.github.io/styleguide/jsguide.html#formatting-function-arguments